sov-modules/sov-modules-impl/bank/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+mod call;
+mod genesis;
+mod query;
+use sov_modules_api::Error;
+use sov_modules_macros::ModuleInfo;
+use sov_state::WorkingSet;
+
+type Amount = u64;
+
+#[derive(borsh::BorshDeserialize, borsh::BorshSerialize, Debug, PartialEq)]
+pub struct Coins<Address: sov_modules_api::AddressTrait> {
+    amount: Amount,
+    token_address: Address,
+}
+
+#[derive(borsh::BorshDeserialize, borsh::BorshSerialize, Debug, PartialEq, Clone)]
+pub struct Token<Address: sov_modules_api::AddressTrait> {
+    name: String,
+    total_supply: u64,
+    balances: sov_state::StateMap<Address, Amount>,
+}
+
+#[derive(ModuleInfo)]
+pub struct Bank<C: sov_modules_api::Context> {
+    #[address]
+    pub address: C::Address,
+
+    #[state]
+    pub names: sov_state::StateMap<String, C::Address>,
```

Comment:
We decided  (offline) to support multiple tokens with the same name. I will remove the mapping. And because of that, `name squatting` is less of an issue.